### ISSUE TITLE
Backport importer-jdbc plugin on official 2.16.x branch

### ISSUE THREAD

**Issue by Gnafu** [geosolutions-it/eumetsat-EOWS#199]
This issue is on hold.
To be decided if we actually need a backport to the 2.16.x or if we are going to just jump to the 2.17.x

**Comment by nmco** [comment]
News on this one @Gnafu ?

**Comment by Gnafu** [comment]
Still on hold, I'm not aware of any ongoing discussion on this.

**Comment by nmco** [comment]
Understood, thx!

**Comment by simboss** [comment]
I unblocked this, but it is low priority.

Move forward if there is **nothing** else to do.

**Comment by imranrajjad** [comment]
### update

I have cherry-picked the commits  listed on https://github.com/geoserver/geoserver/commits/master/src/community/importer-jdbc, however I have noticed that the tests are running on master and failing on backport. Could I be missing something? perhaps like some commits that need to be cherry-picked and have file changes in API outside the module folder.

**Comment by nmco** [comment]
If I'm not mistaken @imranrajjad the ``importer-jdbc plugin`` doesn't exists at all in 2.16.x, if there is a lot of commits PRs touching it, you should not try to cherry-pick, but instead take plugin directory and changes on relevant pom's  and put it in 2.16.x and do a single commit, did you try this?

**Comment by nmco** [comment]
What the status on this @imranrajjad ?

**Comment by imranrajjad** [comment]
@nmco Blocked

Copied the folder from master, and got compile errors. Surely there are some commits that pre-requisite to this backport

**Comment by nmco** [comment]
Can you provide a bit more details @imranrajjad? What compile error do you have?

**Comment by imranrajjad** [comment]
@nmco correction, it compiles but the tests fail

```
[ERROR] Failures:
[ERROR]   H2ImportControllerTest.testJdbcImportStore:18
Expected: an instance of org.geoserver.importer.jdbc.JDBCImportStore
     but: <org.geoserver.importer.MemoryImportStore@792eeae6> is a org.geoserver.importer.MemoryImportStore
[ERROR]   H2ImportTaskControllerTest.testJdbcImportStore:17
Expected: an instance of org.geoserver.importer.jdbc.JDBCImportStore
     but: <org.geoserver.importer.MemoryImportStore@4ef7145f> is a org.geoserver.importer.MemoryImportStore
[ERROR]   H2ImportTransformTest.testJdbcImportStore:17
Expected: an instance of org.geoserver.importer.jdbc.JDBCImportStore
     but: <org.geoserver.importer.MemoryImportStore@1f664881> is a org.geoserver.importer.MemoryImportStore
[ERROR]   H2ImporterDataTest>ImporterDataTest.testImportUnknownFile:374 expected:<1> but was:<0>
[ERROR]   H2ImporterDataTest>ImporterDataTest.testImportUnknownFileIndirect:387 expected:<1> but was:<0>
[ERROR]   H2ImporterDataTest.testJdbcImportStore:17
Expected: an instance of org.geoserver.importer.jdbc.JDBCImportStore
     but: <org.geoserver.importer.MemoryImportStore@54ff871a> is a org.geoserver.importer.MemoryImportStore
[ERROR]   H2ImporterIntegrationTest.testJdbcImportStore:18
Expected: an instance of org.geoserver.importer.jdbc.JDBCImportStore
     but: <org.geoserver.importer.MemoryImportStore@4b7cb8a0> is a org.geoserver.importer.MemoryImportStore
[ERROR]   H2ImporterTest.testJdbcImportStore:17
Expected: an instance of org.geoserver.importer.jdbc.JDBCImportStore
     but: <org.geoserver.importer.MemoryImportStore@4254821c> is a org.geoserver.importer.MemoryImportStore
[ERROR] Errors:
[ERROR]   H2ImporterDataTest>ImporterDataTest.testImportGML2WithSchema:710 » FileNotFound
[ERROR]   H2ImporterDataTest>ImporterDataTest.testImportGML3WithSchema:729 » FileNotFound
[ERROR]   H2JDBCImportStoreTest>AbstractJDBCImportStoreTest.testAdd:109->AbstractJDBCImportStoreTest.runAddTest:122 » NoClassDefFound
[ERROR]   H2JDBCImportStoreTest>AbstractJDBCImportStoreTest.testSave:154->AbstractJDBCImportStoreTest.runAddTest:122 » NoClassDefFound
[INFO]
[ERROR] Tests run: 117, Failures: 8, Errors: 4, Skipped: 9
```

**Comment by Gnafu** [comment]
@imranrajjad you can have a look at the https://github.com/geosolutions-it/geoserver/tree/EOWS-158 branch 
The back-port was already done there so it may help you understanding what's going wrong.

**Comment by simboss** [comment]
@imranrajjad forget about this one and focus on the authentication work

@nmco I would pass this one to @taba90

**Comment by taba90** [comment]
pull request https://github.com/geoserver/geoserver/pull/4076

**Comment by nmco** [comment]
Merged thx @taba90, looks like the referenced issue is unrelated: https://osgeo-org.atlassian.net/browse/GEOS-9398, nobody will die but next time we need to be careful.
